extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/vertx/VertxHttpServerMetrics.java

```diff
@@ -77,12 +77,14 @@
     @Override
     public MetricsContext responsePushed(MetricsContext socketMetric, HttpMethod method, String uri,
             HttpServerResponse response) {
-        String path = VertxMetricsTags.parseUriPath(matchPatterns, ignorePatterns, uri);
-        if (path != null) {
-            registry.counter(nameHttpServerPush,
-                    Tags.of(VertxMetricsTags.uri(path, response.getStatusCode()), VertxMetricsTags.method(method),
-                            VertxMetricsTags.outcome(response), VertxMetricsTags.status(response.getStatusCode())))
-                    .increment();
+        if (socketMetric != null) {
```

Comment:
It's minor and you don't have to fix it this time but we usually recommend to do:
```
if (socketMetric == null) {
    return socketMetric; // or return null
}
```
rather than nesting ifs.

It's far more readable to just take care of the exceptional cases at the beginning of your method and then do your logic.

Reply:
I actually thought of doing that first as I prefer positive checks, too. Give me a few minutes

Reply:
Yeah it's not really about positive checks, I would have said the same thing if it would have been a negative one. The idea is to keep the normal flow as simple as possible by evacuating special cases first.

Reply:
I simplified a few, and clarified javadoc re: null params & return values.